jina/helper.py

```diff
@@ -503,6 +503,11 @@ def valid_yaml_path(path: str, to_stream: bool = False):
                                 ' or a valid file path, or a supported class name.' % path)
 
 
+def valid_docker_image(image: str):
+    # TODO: Think a better way, it is a little dangerous
```

Comment:
@hanxiao , we need a way to better identify wether `uses` refers to a **docker image**

Reply:
can we merge this function with `valid_yaml_path`?

Reply:
it is not so automatic, because `valid_yaml_path`  would be called with 'uses_internal` I think

Reply:
Actually valid_docker_image returns a boolean while valid_yaml_path actually returns a processed object; so different objects.

Reply:
@YueLiu-jina , we also can apply negative logic, and that would save us from early checking it is a valid image. I am not a fan of applying it, but in this case it can save us from trouble. Instead of checking it is a **valid_docker_image**, we can check it is not a **valid_yaml_path**.  And since `uses` should be one OR another it should work, I may also change the naming of `valid_yaml_path` to something that shows that is not only `yaml_paths`

Reply:
@JoanFM I agree with this logic. But I still feel a *valid_image_path* function is needed. If we have this function, when, say, one intends to use yaml path but make a typo, instead of treating it as a docker image and waiting docker to react, we can raise a specific error and maybe terminate the flow, and this will save lots of time. Also for the future, maybe some other formats will be used, and then this one or another won't work.

Reply:
you have a good point there.